### DISCUSSION TITLE
Feature Request: Import Findings from Other Sources and Generate Reports via API

### DISCUSSION THREAD

**Discussion by sarinhacarbon** [Syslifters/sysreptor#44] [General]
Hello,

I have been really enjoying using sysreptor and I think it has a lot of potential. Recently, I had an idea that could further expand its capabilities.

It would be helpful if we could import findings from other sources, such as DefectDojo, directly into sysreptor. This would allow us to leverage the tools and features that we are already using, while still taking advantage of the benefits of sysreptor.

In addition, it would be great if we could generate reports directly from sysreptor via API. This would allow us to further automate our processes and improve efficiency.

Here is an example of how I envision this feature:

1. Provide a sysreptor API that allows for the import of findings from an external source.
2. The user can then use this API to send findings from DefectDojo (or another source) to sysreptor.
3. Sysreptor processes these findings and integrates them into the existing report.
4. The user can then generate a report via API.

I hope this suggestion is helpful and I am looking forward to seeing what you think of it.

Thanks for your time and for the excellent work you are doing with sysreptor!

Best regards,

**Comment by aronmolnar** [comment]
Hi,
Thank you for this idea and I'm happy to hear you like SysReptor.

We plan to implement an API in combination with a CLI tool with a plugin-system. This would allow you to create a plugin to fetch data from external sources (like DefectDojo, or also from tool outputs) and to push them to your report.
